File: api-reference/examples/node/deployment.js
const { NexlayerClient } = require('@nexlayer/sdk');

// Initialize the client
const client = new NexlayerClient({
  // Load token from environment variable
  sessionToken: process.env.NEXLAYER_SESSION_TOKEN
});

async function deployApplication() {
  try {
    // Example YAML configuration
    const yamlConfig = `
application:
  name: my-ai-app
  pods:
    - name: webapp
      image: "your-username/my-app:v1.0.0"
      path: "/"
      servicePorts:
        - 3000
    `;

    // Start the deployment
    const deployment = await client.startDeployment(yamlConfig);
    console.log('Deployment started:', deployment.url);

    // Monitor pod status
    const status = await client.getPodsStatus(deployment.applicationName);
    console.log('Pod status:', status.pods);

    // Extend deployment if needed
    const extension = await client.extendDeployment(deployment.applicationName);
    console.log('Deployment extended:', extension.message);
  } catch (error) {
    console.error('Deployment error:', error.message);
    process.exit(1);
  }
}

deployApplication(); 

File: api-reference/examples/python/deployment.py
from nexlayer import NexlayerClient
import os

def deploy_application():
    try:
        # Initialize client with token from environment
        client = NexlayerClient(
            session_token=os.environ['NEXLAYER_SESSION_TOKEN']
        )

        # Example YAML configuration
        yaml_config = """
application:
  name: my-ai-app
  pods:
    - name: webapp
      image: "your-username/my-app:v1.0.0"
      path: "/"
      servicePorts:
        - 3000
        """

        # Start the deployment
        deployment = client.start_deployment(yaml_config)
        print(f"Deployment started: {deployment.url}")

        # Monitor pod status
        status = client.get_pods_status(deployment.application_name)
        print(f"Pod status: {status.pods}")

        # Extend deployment if needed
        extension = client.extend_deployment(deployment.application_name)
        print(f"Deployment extended: {extension.message}")

    except Exception as e:
        print(f"Deployment error: {str(e)}")
        exit(1)

if __name__ == "__main__":
    deploy_application() 